main.cpp

```diff
@@ -68,46 +85,64 @@ struct Speaker
         float fMin;
         float fMax; 
 
-        void limitInputLevel( float maxIncomingPeakLevel = 10);
+        void limitInputLevel( float maxIncomingPeakLevel = 10 );
         void bandPass( float minFreqAllowed = 20, float maxFreqAllowed = 20000 );
     };
 
-    //1)active design can amplify a line level signal
     Signal amplifySignal( Signal signalToModify );
-    //2)can split the signal in 2 bands at crossover frequency
     Signal splitSignal( Signal signalToModify );
-    //3)port can boost the low frequencies
     Signal boostLowFreq( Signal signalToModify );
 
     Signal signal;
-    
 };
-/*
-2)jumpstarter
-5 properties:
-    1)current rating in amperes
-    2)battery technology
-    3)weight
-    4)short circuit protection
-    5)charge display leds
-3 things it can do:
-    1)jumpstart a car 
-    2)power devices trough USB
-    3)display the battery charge level
- */
+
+Speaker::Signal Speaker::amplifySignal( Signal signalToModify )
+{
+    signalToModify.level += 3;
+    return signalToModify;
+}
+
+Speaker::Signal Speaker::splitSignal( Signal signalToModify )
+{
+    signalToModify.fMax = fc;       //will just return the woofer signal range, can't return 2 signals yet
+    return signalToModify;
+}
+
+Speaker::Signal Speaker::boostLowFreq( Signal signalToModify )
+{
+    signalToModify = splitSignal( signalToModify );
+    signalToModify.level += 1;
+    return signalToModify;
+}
+
+void Speaker::Signal::limitInputLevel( float maxIncomingPeakLevel )
+{
+    if( level > maxIncomingPeakLevel )
+    {
+        level = maxIncomingPeakLevel;
+    };
+}
+
+void Speaker::Signal::bandPass( float minFreqAllowed, float maxFreqAllowed )
+{
+    if( fMin < minFreqAllowed )
+    {
+        fMin = minFreqAllowed;
+    };
```

Comment:
semicolons are used when ending a struct or variable initialization.  they are not used after a function block.
- [ ] revise